lib/fs/metadata.go

```diff
@@ -125,7 +128,7 @@ func (m Metadata) Marshal(out io.Writer) {
 	enc.EncodeString(magic_UTF8, "t") // type
 	enc.EncodeInt(int64(m.Typeflag))
 	enc.EncodeString(magic_UTF8, "m") // mode -- note this is *not* `os.FileMode`, it's just the perm bits
-	enc.EncodeInt(m.Mode)
+	enc.EncodeInt(m.Mode & 07777)
```

Comment:
Ugh.  I'm going to claim that the reason this is badly commented is because the responsibility for these bits is too scattered -- it already is commented elsewhere, just not here.  Which is, yes, absolutely still a problem.  "Suffer no jank" seems to apply in force; listening to that smell would have caused this to be noticed a long time ago -.-

I'm becoming increasingly tempted to break the `type Metadata tar.Header` alias and make our own struct, because though the `tar.Header` fields are *mostly* what we want... the edge cases are Extremely Irritating, both in code to handle them, and doubly so to document.  :soon:?